design/ceph/ceph-external-cluster.md

```diff
@@ -121,3 +121,39 @@ they could simply create them with unique CRD names.
 
 Assuming the object store can be shared with the external cluster, similarly to pools, the owner of the object store is the external cluster.
 If the local cluster attempts to change the pool settings such as replication, they will be ignored.
+
+## Monitoring (prometheus)
+
+Rook already creates and injects service monitoring configuration, consuming what the ceph-mgr prometheus exporter module generates.
+This enables the capability of a Kubernetes cluster to gather metrics from the external cluster and feed them in Prometheus.
+
+The idea is to allow Rook-Ceph to connect to an external ceph-mgr prometheus module exporter.
+
+1. Enhance external cluster script:
+    1. the script tries to discover the list of managers IP addresses
+    2. if provided by the user, the list of ceph-mgr IPs in the script is accepted via the new `--prometheus-exporter-endpoint` flag
+
+2. Add a new entry in the monitoring spec of the `CephCluster` CR:
+
+```go
+// MgrPrometheusExporterExternalEndpoints point to existing Ceph prometheus exporter endpoints
+MgrPrometheusExporterExternalEndpoints []v1.EndpointAddress `json:"mgrPrometheusExporterExternalEndpoints,omitempty"`
+}
+```
+
+So the CephCluster CR will look like:
+
+```yaml
+monitoring:
+  # requires Prometheus to be pre-installed
+  enabled: true
+  mgrPrometheusExporterExternalEndpoints:
+    - ip: "192.168.0.2"
```

Comment:
These endpoints would be the same for dashboard or any other mgr module, right? If it's the same as other mgr modules, how about naming it `externalMgrEndpoints`?

Reply:
Yes, they will be the same but only the Prometheus one is used (internally through the port). So `externalMgrEndpoints` seems a bit generic to me. However, we are inside the `monitoring` section already...

Reply:
But I think it's a good point, what do you think of adding `externalMgrEndpoints` in the `mgr` section instead?

Reply:
In retrospect, the `monitoring` likely should have been included underneath the `mgr` section since they are closely related. Because of the overlap, the setting could go either place. Since the endpoint will only be used for monitoring, it seems better to leave it under `monitoring` as `externalMgrEndpoints`.